### PR TITLE
Add User & Device info

### PR DESCRIPTION
**Pull request by TimeEngineer**

### CODE REVIEW

**Comment by TimeEngineer** [comment]
I would like to add the binding, but some change on DateTime would be apprecied :smile:

File: src/local_device.rs [hunk 1]
```diff
@@ -275,3 +275,142 @@ impl LocalDevice {
...
+
+    fn __repr__(&self) -> PyResult<String> {
+        Ok(format!("{:?}", self.0))
+    }
+
+    fn __richcmp__(&self, other: &Self, op: CompareOp) -> bool {
+        match op {
+            CompareOp::Eq => self.0 == other.0,
+            CompareOp::Ne => self.0 != other.0,
+            CompareOp::Lt => self.0 < other.0,
```
Comment (touilleMan) [inline src/local_device.rs:387]: Why do we need the full range of comparison here ? (given we only have eq/ne for UserInfo)

**Review by touilleMan — COMMENTED** [review]
Seems like a good start, now this should remove the Python implementation of UserInfo/DeviceInfo...

File: src/local_device.rs (hunk 1 above)
Comment (TimeEngineer) [reply src/local_device.rs:387]: There is a sort in tests

Comment (touilleMan) [reply src/local_device.rs:390]: Then I think we should also implement this for the UserInfo (given it should be trivial do to and it's probably interesting to have deterministic order when testing/displaying)

**Review by touilleMan — APPROVED** [review]

**Pull request merged**